### ISSUE TITLE
Errors don't contain location information

### ISSUE THREAD

**Issue by davej** [breuleux/gulp-earl#3]
When compiling with gulp, errors currently look like:

```
ERROR: E.syntax.undeclared: Undeclared variable: ||
```

When working with 10's of files then it can be difficult to track down where the offending code is. Is it possible to log error output that is more similar to the type of output that the `earl` command gives:

```
E.syntax.undeclared: Undeclared variable: ||

/Users/dave/{snip}/services/eg/tunnel.eg 33:38-39
 33:   subdomain = server.public-subdomain || server.local-subdomain
```

**Comment by breuleux** [comment]
Yeah, I should fix that. The issue is that the location information is not formatted where the error is thrown (the error object only contains the AST node(s) for the location, so the catcher can choose how to format it), but `gulp-earl` doesn't format it.

**Comment by davej** [comment]
Yeah, I've quickly edited the plugin so that the filename (from `err.location.source.url`) is added to the error at least, it's good enough for my purposes at the moment.

**Comment by TamsynUlthara** [comment]
I factored out the error formatting code in my local copy of Earl Grey's [`run.eg`](https://github.com/breuleux/earl-grey/blob/master/src/run.eg) into its own function, which I've been using in my local `gulp-earl` to good effect. I'll see about submitting a PR to [`earl-grey`](https://github.com/breuleux/earl-grey) of the former, and if it gets accepted I'll submit a PR here for the latter.
